axiom/datasets.py

```diff
@@ -28,20 +32,104 @@ class IngestStatus:
     wal_length: int
 
 
+@dataclass
+class Dataset:
+    """Represents an Axiom dataset"""
+
+    id: str = field(init=False)
+    name: str
+    description: str
+    who: str
+    created: str
+
+
+@dataclass
+class DatasetCreateRequest:
+    """Request used to create a dataset"""
+
+    name: str
+    description: str
+
+
+@dataclass
+class DatasetUpdateRequest:
+    """Request used to update a dataset"""
+
+    description: str
+
+
+class ContentType(Enum):
+    JSON = "application/json"
+    NDJSON = "application/x-ndjson"
+    CSV = "text/csv"
+
+
 class DatasetsClient:  # pylint: disable=R0903
     """DatasetsClient has methods to manipulate datasets."""
 
     session: Session
 
-    def __init__(self, session: Session):
+    def __init__(self, session: Session, logger: Logger):
         self.session = session
+        self.logger = logger
 
-    def ingest(self, dataset: str, events: List[dict]) -> IngestStatus:
+    def ingest(
+        self, dataset: str, events: List[dict], contentType: str = ContentType.NDJSON
```

Comment:
The other clients have `ingest` which takes a kind of reader and a content type and a content encoding flag _and_ an ingest_events which takes events and always encodes as gzipped ndjson, would be great to have something similar here.

Reply:
Ok, I implemented the same now. only part missing is using the ingest options.

Reply:
completed, please check.